modules/environment/login/login-inner.nix

```diff
@@ -76,9 +76,10 @@ writeText "login-inner" ''
   # Workaround for https://github.com/NixOS/nix/issues/1865
   export NIX_PATH=nixpkgs=$HOME/.nix-defexpr/channels/nixpkgs/:$NIX_PATH
 
-  if [ "$#" -eq 0 ]; then
-    exec /usr/bin/env bash
+  __NOD_SHELL="$(getent passwd ${config.user.userName} | cut -d : -f 7)"
```

Comment:
Leads to `getent: command not found` for me for

1) post-installation (logging in again helps)
2) post-shell-change (fatal)

Reply:
Oh thats strange.. It did work for me, let me investigate it a bit.

Reply:
Removed the getent call completely. Was unneccessary on second look ;)

Reply:
But can you tell me, what this if condition means?

`if [ "$#" -eq 0 ]; then`

Reply:
'if called with zero arguments'

Reply:
Yes, that is clear to me, but what does is mean in this context? Under which circumstances are the arguments set? Why did you choose to use this condition in the first place?

Reply:
Ahh, right. That's a very important (to me) undocumented feature.

`login` (and `login-inner`) can be called directly from other parts of Android to run something that's managed by nix-on-droid. Then login enters proot, the command gets plumbed up to this place, the path to the command would resolve to the one valid inside proot, and, finally, it'll get get executed with the provided arguments.

Please keep it, I need it for my Bluetooth keyboard remapper. =)

Reply:
Ah okay, I will add a small comment in there to explain it